ca_gazebo/src/color_sensor_plugin.cpp

```diff
@@ -0,0 +1,131 @@
+#include <gazebo/common/Plugin.hh>
+#include <ros/ros.h>
+#include "ca_gazebo/color_sensor_plugin.h"
+
+#include "gazebo_plugins/gazebo_ros_camera.h"
+
+#include <string>
+
+#include <gazebo/sensors/Sensor.hh>
+#include <gazebo/sensors/CameraSensor.hh>
+#include <gazebo/sensors/SensorTypes.hh>
+
+
+#include "std_msgs/Bool.h"
+
+
+namespace gazebo
+{
+  // Register this plugin with the simulator
+  GZ_REGISTER_SENSOR_PLUGIN(GazeboRosColor)
+
+  ////////////////////////////////////////////////////////////////////////////////
+  // Constructor
+  GazeboRosColor::GazeboRosColor():
+  _nh("color_sensor_plugin"),
+  _fov(6),
+  _range(10)
+{
+
+}
+  ////////////////////////////////////////////////////////////////////////////////
+  // Destructor
+  GazeboRosColor::~GazeboRosColor()
+  {
+    ROS_DEBUG_STREAM_NAMED("camera","Unloaded");
+  }
+
+  void GazeboRosColor::Load(sensors::SensorPtr _parent, sdf::ElementPtr _sdf)
+  {
+    // Make sure the ROS node for Gazebo has already been initialized
+    if (!ros::isInitialized())
+    {
+      ROS_FATAL_STREAM("A ROS node for Gazebo has not been initialized, unable to load plugin. "
+        << "Load the Gazebo system plugin 'libgazebo_ros_api_plugin.so' in the gazebo_ros package)");
+      return;
+    }
+
+    CameraPlugin::Load(_parent, _sdf);
+    // copying from CameraPlugin into GazeboRosCameraUtils
+    this->parentSensor_ = this->parentSensor;
```

Comment:
Are all these parameters being used?

Reply:
Yes, removing one of them makes the plugin misbehave